connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2807,6 +2807,38 @@ object functions {
   // Misc functions
   //////////////////////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * Returns the current catalog.
+   *
+   * @group misc_funcs
+   * @since 3.5.0
+   */
+  def current_catalog(): Column = Column.fn("current_catalog")
+
+  /**
+   * Returns the current database.
+   *
+   * @group misc_funcs
+   * @since 3.5.0
+   */
+  def current_database(): Column = Column.fn("current_database")
+
+  /**
+   * Returns the current schema.
+   *
+   * @group misc_funcs
+   * @since 3.5.0
+   */
+  def current_schema(): Column = current_database()
```

Comment:
I'm afraid. If the current_schema have the different output from current_database in future(means current_schema doesn't reuse CurrentDatabase and create a new expression CurrentSchema), the behavior will not change. Personally, I think `Column.fn("current_schema")` is better.

Reply:
this only works for Connect Clients. For vanilla Spark (both scala and python), it will still be unaware of such kind of changes, since the expression is directly built in the functions.

To prevent functions from such behavior changes, I guess what we can do is adding more parity tests to compare function with `selectExpr`

Reply:
WDYT? @HyukjinKwon

Reply:
friendly ping @HyukjinKwon

Reply:
we can call `call_udf` instead?

Reply:
I don't mind either way. If we will have `CurrentSchema`, we will likely take a look at `functions.scala` together anyway.

Reply:
> we can call `call_udf` instead?

it is a good idea, but inconsistent with existing implementations.

I am fine to change to directly invoking the `current_schema` to keep in line with other PRs